jupyter_book/book_template/content/guide/03_build.md

```diff
@@ -42,16 +42,52 @@ From here, you have **two options**
 
    Building your book's site locally lets you preview your book locally before you
    push it online. It also gives you a bit more functionality than using
-   GitHub Pages to build your book. However, it also requires you to install Ruby.
+   GitHub Pages to build your book. However, it also requires you to install
+   Docker (a containerization platform) _or_ Ruby (an open source programming language).
    If you'd like to build your site locally then jump to the next section.
 
-## Build the book's site HTML locally (optional)
+## Build the book's site HTML locally
 
-If you'd like to build your book's site HTML locally, you'll need to first
-install the necessary dependencies and then build the HTML. Follow these
-steps to do so.
+Once you've generated the markdown for your notebooks and installed the
+necessary dependencies. You are ready to build your site HTML.
+
+Ensure that your notebooks have been converted to markdown, there should be a
+collection of them in `_build/`.
+
+In order to locally build your site's HTML with Jekyll, you'll need to either install
+Docker or Ruby.
+
+In our experience, we've found that Docker provides an easier installation for most systems.
+We therefore generally recommend that you use the provided Docker image.
+To learn more about Docker using Docker, please see the
+[Docker for scientists guide](https://neurohackweek.github.io/docker-for-scientists/)
+
+### Building your site locally with Docker
+
+First, you'll need to make sure you have Docker installed.
+There are [installation instructions for each operating system](https://hub.docker.com/search/?type=edition&offering=community)
+to guide you through this process.
+
+Once Docker is available on your system, you can build the image locally with:
+
+```bash
+docker build -t jupyter-book /PATH/TO/YOUR/CONTENT
+```
 
-### Install the dependencies to preview your book locally
+You can then access this image with:
+
+```bash
+docker run --rm --security-opt label:disable  \
+   -v /PATH/TO/YOUR/CONTENT:/srv/jekyll \
```

Comment:
does this need to be the full path? Because if I try `.` when I am in the current folder, it tells me:

```sh
docker: Error response from daemon: create .: volume name is too short, names should be at least two alphanumeric characters.
```

and when I try the full path, it tells me:

```sh
Could not find public_suffix-2.0.5 in any of the sources
Run `bundle install` to install missing gems.
```

Reply:
It does need to be the full path ! Or you can use the `$PWD` variable.

And in my experience that means you need to delete your `Gemfile.lock`. Are you using the one in this PR ?

Reply:
Indeed, I had forgotten to copy the `Gemfile.lock` from this PR. It worked. Thanks :)

Reply:
So practically speaking, one would need to first upgrade their jupyter-book to use this, no?

Reply:
I think we need to cut a new release soon because of #120 so hopefully this PR will make it in !

Or you _should_ be able to delete your local `Gemfile.lock`, copy in the `Dockerfile`, and run this command. If you want to try, would love feedback on it !

Reply:
Correct. But even then, existing books cannot use this unless they upgrade (?) Or is there a check somewhere so that the books are upgraded also when one starts using a new version?

Reply:
Sorry, I think I'm missing something. Do you mean the new versions listed in the `Gemfile.lock` or a new version of Jupyter Book itself ?

Reply:
So, if I understand correctly, there are two copies of `Gemfile.lock` -- one in the jupyter-book repository itself and the second which is in the book I created, say my-book. Thus you have both jupyter_book/book_template/Gemfile.lock and my-book/Gemfile.lock. So, even if jupyter_book/book_template/Gemfile.lock is upgraded, my-book/Gemfile.lock will not be updated unless you run the command "jupyter-book upgrade my-book"

Reply:
Ah, ok, thanks for clarifying ! Yes, because your book was made with an older version of Jupyter Book you'll need to update the files in them. I don't think it's worth it to implement an "upgrade" command, at least for now (if the changes become more substantial in the future then it's worth thinking about !). But I can add to the FAQ that if you see the above error, to try deleting your local `Gemfile.lock` ?

Reply:
Oh but you don't have to implement anything ! The feature [already exists](https://jupyter.org/jupyter-book/guide/04_faq.html). I was just suggesting adding a note above.

Reply:
:tada: nice ! Sorry, in my head that didn't update the Gemfiles, but looking again I think it does. Yes, definitely think it makes sense to add a note :smile_cat:

Reply:
@jasmainak I think this is resolved with the new note about updating the books -- let me know if you think it should be clearer !